bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -673,7 +673,7 @@ bool ReplicaImp::tryToSendPrePrepareMsg(bool batchingLogic) {
       time_to_collect_batch_ = MinTime;
     }
   } else {
-    auto builtReq = buildPrePrepareMessage();
+    auto builtReq = buildPrePrepareMessageByRequestsNum(config_.maxNumOfRequestsInBatch);  // hanan
```

Comment:
I'll take a look

Reply:
I added the scoped timers but and reused the code in a little different way than you suggested since buildPrePrepareMessage calls finishAddingRequestsToPrePrepareMsg with requiredRequestsNum = 0, and if I'd just call buildPrePrepareMessageByRequestsNum from buildPrePrepareMessage with a large limit (which I don't like anyway) it'd send the wrong requiredRequestsNum to finishAddingRequestsToPrePrepareMsg. so I changed  buildPrePrepareMessage to call buildPrePrepareMessageByRequestsNum with 0 which means "no limit" and changed the condition in the while loop to meet both cases. 

I think this is more dangerous than keeping the duplicated code so please review this again closely... thanks.